test/main.test.js

```diff
@@ -129,4 +129,49 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    var timeout = 50;
+    // Concurrent execution number 1 of 2
+    var execution1 = new Promise(function execution1(outerResolve, reject) {
```

Comment:
Considering that the code building execution1 and execution2 differ only in one parameter (the value stored in the context), the code would be much easier to read if the creation of execution promise was refactored into a shared function.

```js
Promise.all([
  runTestWithValue('test-value-1'),
  runTestWithValue('test-value-2'),
]);

function runTestWithValue(value) {
  LoopBackContext.runInContext(...);
}
```